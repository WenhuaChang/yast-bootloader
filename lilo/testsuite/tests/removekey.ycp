/*
 * Authors:
 *  Unknown <unknown@addres.xx>
 *
 * $Id$
 *
 */

{
    include "testsuite.ycp";
    import "Pkg";

    // minimal "don't care" SCR data to work with the constructor
    map READ = $[
	"target": $[
	    "tmpdir": "/tmp",
	    ],
	"etc" : $[
            "install_inf" : $[
                "Locale" : "en_EN",
                "InstMode" : "cd",
		],
	    ],
	"sysconfig" : $[
            "console" : $[
                "CONSOLE_FONT" : "lat2-16.psfu",
                "CONSOLE_SCREENMAP" : "trivial",
                "CONSOLE_UNICODEMAP" : "",
                "CONSOLE_MAGIC" : "(K",
                "CONSOLE_ENCODING" : "ISO-8859-2",
            ],
        ],
        "probe" : $[
            "disk" : [],
            "boot_arch" : "lilo",
            "architecture" : "i386",
            "has_apm" : true,
            "has_pcmcia" : false,
            "has_smp" : false,
            "system" : [],
            "floppy" : $[
                "manual" : [],
		],
	    "cdrom" : $[
                "manual" : [],
		],
	    ],
	];
    TESTSUITE_INIT ([READ, $[], $[]], 0);

    include "lilo/routines.ycp";

    TEST(``(removekey($["a":"b"], "a")), [], nil);
    TEST(``(removekey($["a":"b", "c":"d"], "x")), [], nil);
    TEST(``(removekey($["a":"b", "c":"d"], "c")), [], nil);
    TEST(``(removekey(removekey($["a":"b", "c":"d"], "c"), "a")), [], nil);

}
