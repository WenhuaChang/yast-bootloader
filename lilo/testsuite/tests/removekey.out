Return	$[]
Return	$["a":"b", "c":"d"]
Return	$["a":"b"]
Return	$[]
